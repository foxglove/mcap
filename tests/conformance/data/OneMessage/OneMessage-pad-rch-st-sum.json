{
  "records": [
    {"type": "Header", "fields": [["library", ""], ["profile", ""]]},
    {
      "type": "Schema",
      "fields": [
        ["data", ["4", "5", "6"]],
        ["encoding", "c"],
        ["id", "1"],
        ["name", "Example"]
      ]
    },
    {
      "type": "Channel",
      "fields": [
        ["id", "1"],
        ["message_encoding", "a"],
        ["metadata", {"foo": "bar"}],
        ["schema_id", "1"],
        ["topic", "example"]
      ]
    },
    {
      "type": "Message",
      "fields": [
        ["channel_id", "1"],
        ["data", ["1", "2", "3"]],
        ["log_time", "2"],
        ["publish_time", "1"],
        ["sequence", "10"]
      ]
    },
    {"type": "DataEnd", "fields": [["data_section_crc", "340798603"]]},
    {
      "type": "Channel",
      "fields": [
        ["id", "1"],
        ["message_encoding", "a"],
        ["metadata", {"foo": "bar"}],
        ["schema_id", "1"],
        ["topic", "example"]
      ]
    },
    {
      "type": "Statistics",
      "fields": [
        ["attachment_count", "0"],
        ["channel_count", "1"],
        ["channel_message_counts", {"1": "1"}],
        ["chunk_count", "0"],
        ["message_count", "1"],
        ["message_end_time", "2"],
        ["message_start_time", "2"],
        ["metadata_count", "0"],
        ["schema_count", "1"]
      ]
    },
    {
      "type": "SummaryOffset",
      "fields": [
        ["group_length", "50"],
        ["group_opcode", "4"],
        ["group_start", "165"]
      ]
    },
    {
      "type": "SummaryOffset",
      "fields": [
        ["group_length", "68"],
        ["group_opcode", "11"],
        ["group_start", "215"]
      ]
    },
    {
      "type": "Footer",
      "fields": [
        ["summary_crc", "646745152"],
        ["summary_offset_start", "283"],
        ["summary_start", "165"]
      ]
    }
  ],
  "meta": {"variant": {"features": ["st", "rch", "sum", "pad"]}}
}
