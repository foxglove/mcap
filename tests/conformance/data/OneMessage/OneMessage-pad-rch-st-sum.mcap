version https://git-lfs.github.com/spec/v1
oid sha256:8f298361b74d6b3581c13d15c234a60089136c73a6308806b19bad233e235677
size 378
