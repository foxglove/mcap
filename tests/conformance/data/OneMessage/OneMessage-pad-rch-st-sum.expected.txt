Header library="" profile=""
Schema data=<040506> encoding=c id=1 name=Example
ChannelInfo id=1 message_encoding=a metadata={foo=bar} schema_id=1 topic=example
Message channel_id=1 data=<010203> log_time=2 publish_time=1 sequence=10
ChannelInfo id=1 message_encoding=a metadata={foo=bar} schema_id=1 topic=example
Statistics attachment_count=0 channel_count=1 channel_message_counts={1=1} chunk_count=0 message_count=1
SummaryOffset group_length=50 group_opcode=4 group_start=165
SummaryOffset group_length=0 group_opcode=13 group_start=261
SummaryOffset group_length=46 group_opcode=11 group_start=215
SummaryOffset group_length=0 group_opcode=10 group_start=261
SummaryOffset group_length=0 group_opcode=8 group_start=261
Footer summary_crc=0 summary_offset_start=261 summary_start=165
